ExtendedColorSchemes/HarmonyPatches/PlayerDataFileManagerSO.cs

```diff
@@ -56,9 +56,17 @@ private static void GetColorSchemes(ColorSchemesSettings colorSchemesSettings, r
                 if (colorScheme.isEditable)
                 {
                     ColorSchemeWithEditableName? savedColorScheme = Plugin.Config.colorSchemes.FirstOrDefault(x => x.colorSchemeId == colorScheme.colorSchemeId);
-                    ColorSchemeWithEditableName colorSchemeWithEditableName = new(savedColorScheme?.colorSchemeName ?? string.Empty,
-                        colorScheme.colorSchemeId, colorScheme.saberAColor, colorScheme.saberBColor,
-                        colorScheme.environmentColor0, colorScheme.environmentColor1, colorScheme.obstaclesColor);
+                    ColorSchemeWithEditableName colorSchemeWithEditableName = new(
+                        savedColorScheme?.colorSchemeName ?? string.Empty,
+                        savedColorScheme?.colorSchemeId ?? colorScheme.colorSchemeId,
+                        savedColorScheme?.saberAColor ?? colorScheme.saberAColor,
+                        savedColorScheme?.saberBColor ?? colorScheme.saberBColor,
+                        savedColorScheme?.environmentColor0 ?? colorScheme.environmentColor0,
+                        savedColorScheme?.environmentColor1 ?? colorScheme.environmentColor1,
+                        savedColorScheme?.obstaclesColor ?? colorScheme.obstaclesColor,
+                        savedColorScheme?.supportsEnvironmentColorBoost ?? colorScheme.supportsEnvironmentColorBoost,
+                        savedColorScheme?.environmentColor0Boost ?? colorScheme.environmentColor0Boost,
+                        savedColorScheme?.environmentColor1Boost ?? colorScheme.environmentColor1Boost);
```

Comment:
That stuff simply doesn't save anything. You should keep the reference to `colorScheme` because the color schemes aren't saved by that point. The whole point of `savedColorScheme` here is to grab manual edits of the json, I am honestly unsure about how to handle the boost colors there...